spec/features/user/registered_user_can_sign_in_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.feature 'An unauthenticated registered user' do
+  let(:user) { create(:user) }
+  let(:role) { create(:registered_user) }
+
+  scenario 'can visit the log in path and sign in' do
+    user.roles << role
+
+    visit login_path
+
+    expect(current_path).to eq('/login')
+
+    fill_in 'session[email]', with: "#{user.email}"
+    fill_in 'session[password]', with: "#{user.password}"
+
+    click_button 'Login'
+
+    expect(current_path).to eq('/dashboard')
+    expect(page).to have_content("Logged in as #{user.full_name}.")
```

Comment:
Just an ambiguity: You have `flash[:notice] = "Logged in as #{@user.first_name} #{@user.last_name}."` in the code, but here you're testing for `"Logged in as #{user.full_name}."`. Will these always match?